commons/src/main/java/com/powsybl/commons/config/PlatformConfigNamedProvider.java

```diff
@@ -0,0 +1,197 @@
+         * field of {@code moduleName} in {@code platformConfig} among the
+         * {@code providers} arguments based on its name.
+         *
+         * @deprecated Use {@link #findDefault} instead
+         *
+         * @return the provider
+         */
+        @Deprecated
+        public static <T extends PlatformConfigNamedProvider> T findDefaultBackwardsCompatible(
+                String moduleName, Class<T> clazz, PlatformConfig platformConfig) {
+            return find(null, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY,
+                    LEGACY_SERVICE_IMPL_NAME_PROPERTY), clazz, platformConfig);
+        }
+
+        /**
+         * Find the provider among the {@code providers} based on its {@code name}, or
+         * if {@code name} is null find the default provider like @{link
+         * findDefaultBackwardsCompatible}
+         *
+         * @deprecated Use {@link #find} instead
+         *
+         * @return the provider
+         */
+        @Deprecated
+        public static <T extends PlatformConfigNamedProvider> T findBackwardsCompatible(String name,
+                String moduleName, Class<T> clazz, PlatformConfig platformConfig) {
+            return find(name, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY,
+                            LEGACY_SERVICE_IMPL_NAME_PROPERTY),
+                    clazz, platformConfig);
+        }
+
+        private static Optional<String> getOptionalFirstProperty(ModuleConfig moduleConfig,
+                List<String> propertyNames) {
+            return propertyNames.stream().map(moduleConfig::getOptionalStringProperty)
+                    .filter(Optional::isPresent).map(Optional::get).findFirst();
+        }
+
+        @SuppressWarnings("unchecked")
+        private static <K, V, T extends V> T alwaysSameComputeIfAbsent(
+                Map<K, V> map, K key,
+                Function<? super K, T> mappingFunction) {
+            // Casting to (T) is safe if we awlays pass the same T argument for a given key
+            return (T) map.computeIfAbsent(key, mappingFunction);
+        }
+
+        private static <T extends PlatformConfigNamedProvider> T find(String name,
+                String moduleName, List<String> propertyNames, Class<T> clazz,
+                PlatformConfig platformConfig) {
+            List<T> providers = alwaysSameComputeIfAbsent(PROVIDERS, clazz,
+                k -> Lists.newArrayList(ServiceLoader.load(clazz)));
+            return find(name, moduleName, propertyNames, providers, platformConfig);
+        }
+
+        // package private for tests
+        static <T extends PlatformConfigNamedProvider> T find(String name,
+                String moduleName, List<String> propertyNames, List<T> providers,
+                PlatformConfig platformConfig) {
+            Objects.requireNonNull(providers);
+            Objects.requireNonNull(platformConfig);
```

Comment:
We discussed about this few days ago... You propose to remove this assertion, because, if the name is not null we don't need platformConfig. Did you change your mind?

Reply:
Yes, I find it more robust to always require non null. This avoids problems when the code "works" when the name is not null, but suddenly name becomes null and it doesn't work anymore.

We can rework the API in the future in a separate PR to not give this API where parameters are used depending on whether other parameters are null